yarpl/include/yarpl/flowable/FlowableObserveOnOperator.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+namespace yarpl {
+namespace flowable {
+namespace detail {
+
+template <typename T>
+class ObserveOnOperatorSubscriber : public yarpl::flowable::Subscriber<T> {
+ public:
+  ObserveOnOperatorSubscriber(
+      Reference<Subscriber<T>> inner,
+      folly::Executor& executor)
+      : inner_(std::move(inner)), executor_(executor) {}
+
+  // called from upstream EB
+  void onSubscribe(Reference<Subscription> s) override {
+    executor_.add(
+        [ this, self = this->ref_from_this(this), s = std::move(s) ]{
```

Comment:
Why not:

`[inner = inner_, s = std::move(s)]() mutable { inner->onSubscribe(std::move(s)); }`

And for all followup ones too.

Reply:
inner_ goes away if all reference to `this` are lost